src/swarm.js

```diff
@@ -107,6 +109,140 @@ module.exports = (common) => {
             done()
           })
         })
+
+        const getConfig = (addresses) => {
+          return {
+            Addresses: {
+              Swarm: addresses,
+              API: null,
+              Gateway: null
+            }
+          }
+        }
+        const getRepoPath = () => {
+          return '/tmp/.ipfs-' + Math.random().toString().substring(2, 8)
+        }
+
+        describe('Shows connected peers only once', () => {
+          it('Connecting two peers with one address each', (done) => {
+            let nodeA
+            let nodeB
+            let nodeBAddress
+            const addresses = ['/ip4/127.0.0.1/tcp/0']
+            const config = getConfig(addresses)
+            series([
+              (cb) => {
+                factoryInstance.spawnNode(getRepoPath(), config, (err, node) => {
+                  expect(err).to.not.exist()
+                  nodeA = node
+                  cb()
+                })
+              },
+              (cb) => {
+                factoryInstance.spawnNode(getRepoPath(), config, (err, node) => {
+                  expect(err).to.not.exist()
+                  nodeB = node
+                  cb()
+                })
+              },
+              (cb) => {
+                nodeB.id((err, info) => {
+                  expect(err).to.not.exist()
+                  nodeBAddress = info.addresses[0]
+                  cb()
+                })
+              },
+              (cb) => {
+                nodeA.swarm.connect(nodeBAddress, (err) => {
+                  expect(err).to.not.exist()
+                  cb()
+                })
+              },
+              (cb) => {
+                // Waiting to make sure nodes are connected
+                setTimeout(cb, 1000)
+              },
+              (cb) => {
+                nodeA.swarm.peers((err, peers) => {
+                  expect(err).to.not.exist()
+                  expect(peers).to.have.length(1)
+                  cb()
+                })
+              },
+              (cb) => {
+                nodeB.swarm.peers((err, peers) => {
+                  expect(err).to.not.exist()
+                  expect(peers).to.have.length(1)
+                  cb()
+                })
+              }
+            ], done)
+          })
+          it('Connecting two peers with two addresses each', (done) => {
+            let nodeA
+            let nodeB
+            let nodeBAddress
+            const configA = getConfig([
+              // Need to explicitly declare the ports as 0 doesn't work with
+              // multiple swarm addresses apparently
+              // Not sure how to deal with the case if the ports are busy, this
+              // is introducing flakyness
```

Comment:
This is really weird to me. See https://github.com/ipfs/interface-ipfs-core/pull/150#issuecomment-325125461

Reply:
Test case to reproduce the issue: https://github.com/libp2p/js-libp2p-swarm/pull/227

Reply:
Add a TODO here and create an issue to track this so that we can merge this PR as it is now.